subset/alliedSwitch/src/main/java/alliedswitch/AlliedTelnetClientSocket.java

```diff
@@ -0,0 +1,425 @@
+        new SuppressGAOptionHandler(true, true, true, true);
+
+    try {
+      telnetClient.addOptionHandler(terminalTypeOptionHandler);
+      telnetClient.addOptionHandler(echoOptionHandler);
+      telnetClient.addOptionHandler(suppressGAOptionHandler);
+    } catch (InvalidTelnetOptionException e) {
+      System.err.println(
+          "Error registering option handlers InvalidTelnetOptionException: " + e.getMessage());
+    } catch (IOException e) {
+      System.err.println("Error registering option handlers IOException: " + e.getMessage());
+    }
+  }
+
+  private String normalizeLineEnding(byte[] bytes, char endChar) {
+    String data = new String(bytes);
+
+    List<Byte> bytesBuffer = new ArrayList<Byte>();
+
+    int countBreak = 0;
+    int countESC = 0;
+
+    for (int i = 0; i < bytes.length; i++) {
+      if (bytes[i] != 0) {
+        switch (bytes[i]) {
+          case 8:
+            // backspace \x08
+            break;
+          case 10:
+            // newLineFeed \x0A
+            countBreak++;
+            bytesBuffer.add((byte) endChar);
+            break;
+          case 13:
+            // carriageReturn \x0D
+            countBreak++;
+            bytesBuffer.add((byte) endChar);
+            break;
+          case 27:
+            // escape \x1B
+            countESC = 2;
+            break;
+          case 33:
+            // character:!
+            break;
+          default:
+            if (countESC == 0) {
+              if (countBreak > 1) {
+                int size = bytesBuffer.size();
+                for (int x = 0; x < countBreak - 1; x++) {
+                  bytesBuffer.remove(size - 1 - x);
+                }
+                countBreak = 0;
+              }
+              bytesBuffer.add(bytes[i]);
+            } else {
+              countESC--;
+            }
+            break;
+        }
+      }
+    }
+
+    String bytesString = "";
+
+    for (Byte byteBuffer : bytesBuffer) {
+      bytesString = bytesString + (char) (byte) byteBuffer;
+    }
+
+    return bytesString;
+  }
+
+  private void readData() {
+    int bytesRead = 0;
+
+    inputStream = telnetClient.getInputStream();
+
+    while (telnetClient.isConnected()) {
+      try {
+        byte[] buffer = new byte[1024];
+
+        bytesRead = inputStream.read(buffer);
+        if (bytesRead > 0) {
+          String rawData = normalizeLineEnding(buffer, '\n');
+          rxQueue.add(rawData);
+          // rxQueue.add(new String(buffer, 0, bytesRead, StandardCharsets.UTF_8));
+        } else {
+          try {
+            Thread.sleep(100);
+          } catch (InterruptedException e) {
+            System.err.println("InterruptedException readData:" + e.getMessage());
+            e.printStackTrace();
+          }
+        }
+      } catch (IOException e) {
+        System.err.println("Exception while reading socket:" + e.getMessage());
+      }
+    }
+  }
+
+  private void gatherData() {
```

Comment:
function is way too long

Reply:
function refactored and split where applicable

Reply:
it's still clocking in at over 140 lines... ?  Or am I looking at the wrong place?